### PR TITLE
[#21] Add codecov to the ci/cd pipeline of the project

### PR DESCRIPTION
**Pull request by luongvo**
https://github.com/nimblehq/flutter_templates/issues/21

## What happened 👀

We need to keep our test coverage under observation. This helps the team to always be aware of new code being added are tested thoroughly before getting out to be used by someone else.

## Insight 📝

- Add codecov setup into `.template` project.
- Add sample CodeCov config for `private` repository

## Proof Of Work 📹

Since we moved the main project into [.template](https://github.com/nimblehq/flutter_templates/tree/develop/.template) folder, we need to add support executing `.template` project CI/CD in following tasks, i.e. https://github.com/nimblehq/flutter_templates/issues/47. Please checkout the last Pow from [this PR](https://github.com/nimblehq/flutter_templates/pull/45) 🙏 :

- CodeCov integration https://app.codecov.io/gh/nimblehq/flutter_templates/commits

image

### CODE REVIEW

File: .template/.github/workflows/test.yml [hunk 1]
```diff
@@ -36,3 +36,12 @@ jobs:
...
         run: flutter test --machine --coverage
+
+      - uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+          files: ./coverage/lcov.info
+          flags: unittests # optional
+          name: Upload coverage to codecov
+          fail_ci_if_error: false
+          verbose: true
```
Comment (luongvo) [inline .template/.github/workflows/test.yml:47]: FYI We will try to execute test and develop CI for `.template` project in following tasks i.e. https://github.com/nimblehq/flutter_templates/issues/47.

File: .template/.github/workflows/test.yml [hunk 2]
```diff
@@ -36,3 +36,12 @@ jobs:
 
       - name: Run widget tests, unit tests.
         run: flutter test --machine --coverage
+
+      - uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+          files: ./coverage/lcov.info
+          flags: unittests # optional
+          name: Upload coverage to codecov
```
Comment (manh-t) [inline .template/.github/workflows/test.yml:45]: @luongvo could you help to re-order `name` on top of others? 😄

**Review by manh-t — COMMENTED** [review]
rest LGTM 👏

File: .template/.github/workflows/test.yml (hunk 2 above)
Comment (luongvo) [reply .template/.github/workflows/test.yml]: @manh-t done

**Review by manh-t — APPROVED** [review]
LGTM 💯

File: .template/codecov.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Sample CodeCov config for private repository
+codecov:
+  bot: luongvo
```
[the remark this answers is not in the document]
Comment (luongvo) [reply .template/codecov.yml:3]: @markgravity It requires an account with access to your code repository. We can create a bot account and invited it in but that's the setup task after initializing the project. I documented it in README https://github.com/nimblehq/flutter_templates/pull/90/files#diff-45205db2b83092d10e34cf43e2ef4a0764c3031d3c51c1e13e12c0f26b576d0bR55-R57

**Review by markgravity — APPROVED** [review]
LGTM 👍

**Pull request merged**